packages/@aws-cdk/custom-resources/lib/aws-custom-resource/aws-custom-resource.ts

```diff
@@ -209,6 +210,14 @@ export class AwsCustomResource extends cdk.Construct implements iam.IGrantable {
       }
     }
 
+    for (const call of [props.onCreate, props.onUpdate, props.onDelete]) {
+      if (call && call.ignoreErrorCodesMatching && call.physicalResourceId?.responsePath) {
+        throw new Error('`PhysicalResourceId.fromResponse` cannot be used along with `ignoreErrorCodesMatching`.');
```

Comment:
Not true for `onDelete`, since that cannot produce outputs. Deletes are safe to ignore errors for.

Actually, only true for `onCreate`, since I would assume `onUpdate` leaves the "Data" the same (though I might be wrong). Though it's probably fine to treat `onCreate` and `onUpdate` the same if that's simpler, and then let's wait for a feature request with a use case where there is need for a distinction between them.

Reply:
```suggestion
        throw new Error('`PhysicalResourceId.fromResponse` cannot be used if you supply `ignoreErrorCodesMatching` in either `onCreate` or `onUpdate`.');
```

Reply:
Wait, I don't get the check location. The error message is about `fromResponse` but the `if` is about `responsePath`.

Reply:
Also wouldn't it make more sense to call `breakIgnoreErrorsCircuit` here?

Reply:
@rix0rrr 

> Actually, only true for onCreate, since I would assume onUpdate leaves the "Data" the same (though I might be wrong).

The problem is that because the error is caught, we submit a "SUCCESS" response to CloudFormation with and empty `Data` key - wouldn't this cause an override to the attributes? 

> Wait, I don't get the check location. The error message is about fromResponse but the if is about responsePath.

Yeah, it is confusing. Checking if `responsePath` exists in the only way to check if `fromResponse` is used (its equivalent). The message specifies `fromResponse` because that is what's exposed to the user, not `responsePath`.

> Also wouldn't it make more sense to call breakIgnoreErrorsCircuit here?

Nice 👍

Reply:
@rix0rrr 

> Not true for onDelete, since that cannot produce outputs. Deletes are safe to ignore errors for.

If `fromResponse` is used on the `onDelete` function, we will attempt to extract the id from the response as well. See https://github.com/aws/aws-cdk/blob/master/packages/%40aws-cdk/custom-resources/lib/aws-custom-resource/runtime/index.ts#L130. I think what you say is true for the `getData` validation, but not for the `fromResponse`. Unless. we change the lambda code.

Reply:
Seems like an oversight, or it never mattered, but `{ Ref }`ing or ` { Fn::GetAtt }`ing a deleted resource doesn't make any sense.

In which case, there's no issue right? Unless the JS code throws, in which case we might have a problem, but if it just evaluates to `undefined`, then meh. CloudFormation will never retrieve the value.

Reply:
> The problem is that because the error is caught, we submit a "SUCCESS" response to CloudFormation with an empty `Data` key - wouldn't this cause an override to the attributes?

Yes it will. We would need to explicitly write code to keep the attrs the same. Maybe it's even intractable, since I don't think CFN passes us the old attributes (only the old properties) so we would need external storage ourselves to keep them.

So forget it, treat `onCreate` and `onUpdate` the same, just `onDelete` is safe to ignore errors on.

Reply:
@rix0rrr 

> In which case, there's no issue right? Unless the JS code throws, in which case we might have a problem, but if it just evaluates to undefined, then meh. CloudFormation will never retrieve the value.

So the thing is that `physicalResourceId` evaluates to `undefined`:

```typescript
if (call.physicalResourceId?.responsePath) {
        physicalResourceId = flatData[call.physicalResourceId.responsePath];
}
```

This alone doesn't throw, but when the request is submitted to CFN, it fails with an `Invalid Physical ID` error. I haven't checked though that this also happens on `onDelete`, my guess is yes, but i will verify.

Reply:
> So forget it, treat onCreate and onUpdate the same, just onDelete is safe to ignore errors on.

👍 + see my comment above

Reply:
@rix0rrr Yeap:

```console
2 | 8:26:06 PM | DELETE_FAILED        | Custom::AWS           | Playground/FakeResource/Resource/Default (PlaygroundFakeResourceBC34BA32) Invalid PhysicalResourceId
	new CustomResource (/Users/epolon/dev/src/github.com/aws/aws-cdk/packages/@aws-cdk/aws-cloudformation/lib/custom-resource.ts:163:21)
	\_ new AwsCustomResource (/Users/epolon/dev/src/github.com/aws/aws-cdk/packages/@aws-cdk/custom-resources/lib/aws-custom-resource/aws-custom-resource.ts:259:27)
	\_ new AwsCdkPlaygroundConstruct (/Users/epolon/dev/src/github.com/aws/aws-cdk-playground/lib/aws-cdk-playground-stack.ts:19:5)
	\_ Object.<anonymous> (/Users/epolon/dev/src/github.com/aws/aws-cdk-playground/bin/aws-cdk-playground.ts:15:1)
	\_ Module._compile (internal/modules/cjs/loader.js:778:30)
	\_ Module.m._compile (/Users/epolon/dev/src/github.com/aws/aws-cdk-playground/node_modules/ts-node/src/index.ts:536:23)
	\_ Module._extensions..js (internal/modules/cjs/loader.js:789:10)
	\_ Object.require.extensions.(anonymous function) [as .ts] (/Users/epolon/dev/src/github.com/aws/aws-cdk-playground/node_modules/ts-node/src/index.ts:539:12)
	\_ Module.load (internal/modules/cjs/loader.js:653:32)
	\_ tryModuleLoad (internal/modules/cjs/loader.js:593:12)
	\_ Function.Module._load (internal/modules/cjs/loader.js:585:3)
	\_ Function.Module.runMain (internal/modules/cjs/loader.js:831:12)
	\_ main (/Users/epolon/dev/src/github.com/aws/aws-cdk-playground/node_modules/ts-node/src/bin.ts:212:14)
	\_ Object.<anonymous> (/Users/epolon/dev/src/github.com/aws/aws-cdk-playground/node_modules/ts-node/src/bin.ts:470:3)
	\_ Module._compile (internal/modules/cjs/loader.js:778:30)
	\_ Object.Module._extensions..js (internal/modules/cjs/loader.js:789:10)
	\_ Module.load (internal/modules/cjs/loader.js:653:32)
	\_ tryModuleLoad (internal/modules/cjs/loader.js:593:12)
	\_ Function.Module._load (internal/modules/cjs/loader.js:585:3)
	\_ Function.Module.runMain (internal/modules/cjs/loader.js:831:12)
```

I suggest we apply the `fromResponse` validation to `onDelete` as well for now. This feels like the safest approach, rather than start changing behavior in the lambda code. 

Later on we can think of maybe having a default value for `onDelete` when `ignoreErrorCodesMatching` and `fromResponse` is used (quite a mouthful, we can probably come up with a nicer solution).

Thoughts?